src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -51,14 +51,8 @@ public static BuildEnvironment Instance
                 }
                 catch (TypeInitializationException e)
                 {
-                    if (e.InnerException != null)
-                    {
-                        // Throw the error that caused the TypeInitializationException.
-                        // (likely InvalidOperationException)
-                        throw e.InnerException;
-                    }
-
-                    throw;
+                    // Throw the error that caused the TypeInitializationException, likely InvalidOperationException.
+                    throw e.InnerException ?? e;
```

Comment:
I think this is actually different, and you should not make this change: `throw e;` and `throw;` have different results, because `throw e;` throws _from this stack location_, while `throw;` _rethrows_ the existing exception with its own stack location.